### PR TITLE
Add GetFileSize to filesystem.h

### PR DESCRIPTION
**Pull request by gsoltis**
Does not include windows port

### CODE REVIEW

File: Firestore/core/src/firebase/firestore/util/filesystem.h [hunk 1]
```diff
@@ -72,6 +72,12 @@ Status RecursivelyDelete(const Path& path);
  */
 Path TempDir();
 
+/**
+ * On success, sets `size` to be the size in bytes of the file specified by
+ * `path`.
+ */
+Status GetFileSize(const Path& path, off_t* size);
```
Comment (wilhuff) [inline Firestore/core/src/firebase/firestore/util/filesystem.h]: How about `StatusOr<off_t>` as a return type?

`TempDir` similarly returns a value but isn't "GetTempDir". Could this drop the "Get" and be `FileSize`?

File: Firestore/core/src/firebase/firestore/util/filesystem_posix.cc [hunk 2]
```diff
@@ -95,6 +95,17 @@ Path TempDir() {
 }
 #endif  // !defined(__APPLE__)
 
+Status GetFileSize(const Path& path, off_t* size) {
+  struct stat st;
```
Comment (wilhuff) [inline Firestore/core/src/firebase/firestore/util/filesystem_posix.cc]: Initialize this so automated tools don't ding us for possibly reading uninitialized memory.

```c++
struct stat st{};
```

File: Firestore/core/src/firebase/firestore/util/filesystem.h (hunk 1 above)
Comment (gsoltis) [reply Firestore/core/src/firebase/firestore/util/filesystem.h]: Done on both counts.

File: Firestore/core/src/firebase/firestore/util/filesystem_posix.cc (hunk 2 above)
Comment (gsoltis) [reply Firestore/core/src/firebase/firestore/util/filesystem_posix.cc]: Done.

**Pull request merged**